src/controllers/AuthController.js

```diff
@@ -224,6 +226,71 @@ class AuthController {
     return ({ email, password });
   }
 
+  /**
+   * Verifies if the token passed is linked to an active user and
+   *  performs pre-checks for staff authorization.
+   * @async
+   * @static
+   * @param {Object} req - The request object.
+   * @param {string} req.get - The function to retrieve header values.
+   * @returns {Promise<Object>} The result of the staff pre-check.
+   * @throws {Error} If there is an error during the pre-check process.
+   */
+  static async staffPreCheck(req) {
```

Comment:
Right me if I'm wrong, but doesn't this perform the same functionality as `checkCurrConn`/`checkConn` above? 🤔